integration_tests/sanity/mount_change_test.go

```diff
@@ -0,0 +1,174 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sanity
+
+import (
+	"io/ioutil"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+
+	"github.com/openebs/node-disk-manager/integration_tests/k8s"
+	"github.com/openebs/node-disk-manager/integration_tests/udev"
+)
+
+var _ = Describe("Mount-point and fs type change detection tests", func() {
+	var k8sClient k8s.K8sClient
+	physicalDisk := udev.NewDisk(DiskImageSize)
+
+	var bdName, bdNamespace, mountPath string
+
+	BeforeEach(setUp(&k8sClient, &physicalDisk, &bdName, &bdNamespace, &mountPath))
+	AfterEach(tearDown(&k8sClient, &physicalDisk))
+
+	It("should pass mount-umount flow", func() {
+		// Mount flow
+		By("Running mount flow")
+		By("Mounting disk", mountDisk(&physicalDisk, mountPath))
+		By("Waiting for etcd update", k8s.WaitForStateChange)
+		By("Verifying bd mount point update in etcd",
+			verifyMountPointUpdate(&k8sClient, bdName, bdNamespace,
+				Equal(mountPath)))
+
+		// Umount flow
+		By("Running umount flow")
+		By("Umounting disk", umountDisk(&physicalDisk))
+		By("Waiting for etcd update", k8s.WaitForStateChange)
+		By("Verifying bd mount point update in etcd",
+			verifyMountPointUpdate(&k8sClient, bdName, bdNamespace,
+				BeEmpty()))
+	})
+})
+
+func setUp(kcli *k8s.K8sClient, disk *udev.Disk, bdName, bdNamespace, mountPath *string) func() {
+	return func() {
+		By("initializing up k8s cli", initK8sCli(kcli))
+		By("creating up ndm daemonset", createNDMDaemonset(kcli))
+		By("setting up fs on disk", setupPhysicalDisk(disk))
+		By("attaching disk", attachDisk(disk))
+		By("generating random mount path", generateMountPath(mountPath))
+		By("waiting for etcd update", k8s.WaitForStateChange)
+		By("verifying disk added to etcd", verifyDiskAddedToEtcd(kcli, disk.Name,
+			bdName, bdNamespace))
+	}
+}
+
+func tearDown(kcli *k8s.K8sClient, disk *udev.Disk) func() {
+	return func() {
+		By("detaching disk", detachDisk(disk))
+		By("destroying ndm daemonset", destoyingDaemonset(kcli))
+	}
+
+}
+
+func generateMountPath(mountPath *string) func() {
+	return func() {
+		mp, err := ioutil.TempDir("", "ndm-integration-tests")
+		Expect(err).ToNot(HaveOccurred())
+		*mountPath = mp
+	}
+}
+
+func initK8sCli(cli *k8s.K8sClient) func() {
+	return func() {
+		kcli, err := k8s.GetClientSet()
+		Expect(err).ToNot(HaveOccurred())
+		*cli = kcli
+	}
+}
+
+func setupPhysicalDisk(disk *udev.Disk) func() {
+	return func() {
+		Expect(disk.CreateFileSystem()).ToNot(HaveOccurred())
+	}
+}
+
+func createNDMDaemonset(cli *k8s.K8sClient) func() {
```

Comment:
there is already a function called createNDMDaemonSet. Since this function also waits for pod to come into running state, can choose a different func name.

Reply:
done